src/library/scala/runtime/PStatics.scala

```diff
@@ -0,0 +1,19 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.runtime
+
+// things that should be in `Statics`, but can't be yet for bincompat reasons
+// TODO 3.T: move to `Statics`
+private[scala] object PStatics {
+  final val VM_MaxArraySize = 2147483645 // == `Int.MaxValue - 2`, hotspot limit
```

Comment:
Is the hotspot limit documented somewhere? If it is, a link would come handy. I there any reason why `Int.MaxValue - 2` is only a comment? Why not writing `val VM_MaxArraySize = Int.MaxValue - 2` instead?

Reply:
I don't know where, if anywhere, the hotspot limit is documented. I used the exact value because I can never remember how much constant folding the compiler does, and I wanted to make sure it was a constant. I remember there being a limit to how much folding the compiler will do

Reply:
I have found this: https://stackoverflow.com/questions/3038392/do-java-arrays-have-a-maximum-size . It seems VM dependent, a value of `Integer.MAX_VALUE - 8` is mentioned in one of the answers: https://stackoverflow.com/a/8381338/16673

Reply:
I could perhaps be convinced to change it to that. I'm more inclined to just pick the lowest value that anyone can find on a VM though. it also doesn't hurt to reduce it later if need be